File: wrappers/dotnet/indy-sdk-dotnet-test/NonSecretsTests/DeleteRecordTagsTest.cs
using Hyperledger.Indy.NonSecretsApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.NonSecretsTests
{
    [TestClass]
    public class DeleteRecordTagsTest : NonSecretsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestDeleteRecordTagsWorks()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tags);
            await CheckRecordFieldAsync(wallet, type, id, "tags", tags);

            await NonSecrets.DeleteRecordTagsAsync(wallet, type, id, "[\"tagName1\"]");

            var expectedTags = "{\"tagName2\": \"5\", \"tagName3\": \"12\"}";
            await CheckRecordFieldAsync(wallet, type, id, "tags", expectedTags);
        }

        [TestMethod]
        public async Task TestDeleteRecordTagsWorksForDeleteAll()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tags);
            await CheckRecordFieldAsync(wallet, type, id, "tags", tags);

            await NonSecrets.DeleteRecordTagsAsync(wallet, type, id, "[\"tagName1\", \"tagName2\", \"tagName3\"]");
            await CheckRecordFieldAsync(wallet, type, id, "tags", tagsEmpty);
        }

        [TestMethod]
        public async Task TestDeleteRecordTagsWorksForNotFoundRecord()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
                NonSecrets.DeleteRecordTagsAsync(wallet, type, id, "[\"tagName1\"]")
            );

        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/non_secrets/DeleteRecordTagsTest.java
package org.hyperledger.indy.sdk.non_secrets;

import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;


public class DeleteRecordTagsTest extends NonSecretsIntegrationTest {

	@Test
	public void testDeleteRecordTagsWorks() throws Exception {
		WalletRecord.add(wallet, type, id, value, tags).get();

		checkRecordField(wallet, type, id, "tags", tags);

		WalletRecord.deleteTags(wallet, type, id, "[\"tagName1\"]").get();

		String expectedTags = "{\"tagName2\": \"5\", \"tagName3\": \"12\"}";
		checkRecordField(wallet, type, id, "tags", expectedTags);
	}

	@Test
	public void testDeleteRecordTagsWorksForNotFoundRecord() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		WalletRecord.deleteTags(wallet, type, id, "[\"tagName1\"]").get();
	}
}